go/libkb/naclwrap.go

```diff
@@ -277,13 +294,59 @@ func (k NaclSigningKeyPair) Sign(msg []byte) (ret *NaclSigInfo, err error) {
 		err = NoSecretKeyError{}
 		return
 	}
+
+	// Version 0 is just over the unprefixed message (assume version 0 if no version present)
+	// Version 1 is the same.
 	ret = &NaclSigInfo{
-		Kid:      k.GetKID().ToBytes(),
+		Kid:      k.GetBinaryKID(),
 		Payload:  msg,
 		Sig:      *k.Private.Sign(msg),
 		SigType:  SigKbEddsa,
 		HashType: HashPGPSha512,
 		Detached: true,
+		Version:  0,
+	}
+	return
+}
+
+func (p SignaturePrefix) hasNullByte() bool {
+	for _, b := range []byte(p) {
+		if b == 0 {
+			return true
+		}
+	}
+	return false
+}
+
+type SignaturePrefix string
+
+func (p SignaturePrefix) Prefix(msg []byte) []byte {
+	prefix := append([]byte(p), 0)
```

Comment:
Your original proposal was for a 20-byte prefix, zero-padded.  Is there a reason you changed?